### ISSUE TITLE
Banuba AR Face Filter SDK API Library for Video Call

### ISSUE THREAD

**Issue by wlvideocalldev** [AgoraIO-Extensions/react-native-agora#689]
I want to integrate Banuba with Agora in React Native. I found this link to integrate Banuba with Agora, but how do I implement this code into Agora for React Native? This is the reference link for integrating Banuba with Agora using the Agora Library Extension: [https://console.agora.io/marketplace/extension/introduce?serviceName=banuba](https://console.agora.io/marketplace/extension/introduce?serviceName=banuba)

**Comment by LichKing-2234** [comment]
you need knowledge about Android and iOS, it same as on React Native.

**Comment by wlvideocalldev** [comment]
In the mentioned reference, to integrate the Banuba extension with Agora, the code snippet is as follows:

```kt
RtcEngineConfig().apply {
    ...
    System.loadLibrary("banuba")
    addExtension(BanubaExtensionManager.EXTENSION_NAME)
    ContextProvider.setContext(mContext)
    ...
}
```

However, in this particular package, Agora is initialized using the Iris API engine. How can I incorporate the addExtension method into the Iris API engine setup?

**Comment by LichKing-2234** [comment]
no need to call `addExtension`

**Comment by wlvideocalldev** [comment]
After I run the initialization, I proceed to enable the extension like this:

<img width="626" alt="Screenshot 2023-08-09 at 14 54 23" src="https://github.com/AgoraIO-Extensions/react-native-agora/assets/125262288/24dbef9d-4318-43cf-9cf8-a94335cf32b5">

But, the result I'm getting is -3

**Comment by wlvideocalldev** [comment]
```sh
callApi RtcEngine_enableExtension {"provider":"Banuba","extension":"BanubaFilter","enable":true,"type":100} {"result":-3}
```

**Comment by LichKing-2234** [comment]
have you call System.loadLibrary("banuba") on native?

**Comment by wlvideocalldev** [comment]
Yes, I have loaded the Banuba library using loadLibrary("banuba"). I used the mapLibraryName function to check whether the library has been fully loaded or not, and I received this output: 'libbanuba.so'.

**Comment by wlvideocalldev** [comment]
@LichKing-2234  Yes, I have loaded the Banuba library using loadLibrary("banuba"). I used the mapLibraryName function to check whether the library has been fully loaded or not, and I received this output: 'libbanuba.so'.

![263163979-e2fcfff1-76cc-45fc-b3e4-52941f6f231c](https://github.com/AgoraIO-Extensions/react-native-agora/assets/125262288/f76546a3-9088-4063-9fac-1350384d549a)

But I'm still getting a result of -3.
